plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -129,9 +129,10 @@ protected function render( $attributes, $content, $block ) {
 			* @since 7.6.0
 			* @param boolean.
 			*/
-			$is_disabled_compatibility_layer = apply_filters( 'woocommerce_disable_compatibility_layer', false );
+			$is_disabled_compatibility_layer   = apply_filters( 'woocommerce_disable_compatibility_layer', false );
+			$is_not_purchasable_single_product = ProductType::SIMPLE === $product_type && ( ! $product->is_in_stock() || ! $product->is_purchasable() );
 
-			if ( ! $is_disabled_compatibility_layer ) {
+			if ( ! $is_disabled_compatibility_layer && ! $is_not_purchasable_single_product ) {
 				ob_start();
 				if ( ProductType::SIMPLE === $product_type ) {
```

Comment:
Nit pick but what do you think about moving this check into the `if` blocks below, like the following? It's just my preference in this case though, as I don't like the combination of a general condition for all product (`$is_disabled_compatibility_layer`) with a specific condition for simple product in a same place.

```suggestion
			if ( ! $is_disabled_compatibility_layer ) {
				ob_start();
				if ( ProductType::SIMPLE === $product_type && $product->is_in_stock() && $product->is_purchasable() ) {
```

Reply:
Oh, yes, much cleaner. Thanks, I implemented it in afc9ff2967eb8cbe19b10d73bb78e63fe26db94b.